eis_toolkit/spatial_analyses/cell_based_association.py

```diff
@@ -0,0 +1,573 @@
+import os
+from numbers import Number
+
+import geopandas as gpd
+import numpy as np
+import pandas as pd
+import rasterio
+from beartype import beartype
+from beartype.typing import List, Optional, Tuple, Union
+from shapely import wkt
+from shapely.geometry import Polygon
+
+os.environ["USE_PYGEOS"] = "0"
+
+
+@beartype
+def cell_based_association(
+    cell_size: int,
+    geodata: List[gpd.GeoDataFrame],
+    output_path: str,
+    column: Optional[List[str]] = [""],
+    subset_target_attribute_values: Optional[List[Union[None, list]]] = [None],
+    add_name: Optional[List[Union[str, None]]] = [None],
+    add_buffer: Optional[List[Union[Number, bool]]] = [False],
```

Comment:
The arguments here that have some lists as default values (such as [None] or ["]) should have booleans as default values. Optional type means that the parameter is either None or the type inside the Optional, now if user gives None things will brake down. Also, if you have these lists as default values and modify these variables (such as column) directly, unwanted and -expected things can happen if these functions are called multiple times. So, I would suggest to change the function signature to this:
```suggestion
def cell_based_association(
    cell_size: int,
    geodata: List[gpd.GeoDataFrame],
    output_path: str,
    column: Optional[List[str]] = None,
    subset_target_attribute_values: Optional[List[Union[None, list]]] = None,
    add_name: Optional[List[Union[str, None]]] = None,
    add_buffer: Optional[List[Union[Number, bool]]] = None,
```

Then, before anything else in the code, you can set the starting values like this:
`if column is None:
        column = [""]`